src/models/highlightedtext.cpp

```diff
@@ -263,9 +305,12 @@ void HighlightedText::setText(const QStringList& text)
     }
 
     m_highlightedLines.reserve(text.size());
```

Comment:
don't we need to clear this container when we setText? otherwise repeated calls will just append to it?